### PR TITLE
Add player_monoids compatibility.

### PR DESCRIPTION
**Pull request by shacknetisp**
Currently stamina interferes with any other physics modifications, including gravity. This PR adds functionality to use [player_monoids](https://github.com/minetest-mods/player_monoids) when that API is available so other mods using the same API are no longer broken.

### CODE REVIEW

File: init.lua [hunk 1]
```diff
@@ -103,33 +102,43 @@ end
...
-	-- Get player physics from 3d_armor mod
-	if armor_mod and armor and armor.def then
-		def.speed = armor.def[name].speed
-		def.jump = armor.def[name].jump
-		def.gravity = armor.def[name].gravity
-	end
+	if player_monoids_mod and player_monoids then
+		if sprinting == true then
+			player_monoids.speed:add_change(player, 1 + SPRINT_SPEED, "stamina:physics")
+			player_monoids.jump:add_change(player, 1 + SPRINT_JUMP, "stamina:physics")
```
Comment (SmallJoker) [inline init.lua:114]: How about gravity?

Comment (shacknetisp) [reply init.lua:114]: stamina doesn't affect gravity, does it?

Comment (SmallJoker) [reply init.lua:114]: Stamina not directly, but stamina supports armor (see armor_mod checks) which might use different gravity values.

Comment (shacknetisp) [reply init.lua:114]: Well we're not overriding any gravity values this way. 3d_armor supports player_monoids for speed and jump too so we don't need to manually insert them in order to remain compatible when using this API.

Comment (SmallJoker) [reply init.lua:114]: Good :+1:

**Comment by SmallJoker** [comment]
#26

**Pull request closed without merging**